cobigen/cobigen-tsplugin/src/main/java/com/devonfw/cobigen/tsplugin/inputreader/TypeScriptInputReader.java

```diff
@@ -0,0 +1,289 @@
+        request.executingExe(Constants.EXE_NAME, this.getClass());
+        request.initializeConnection();
+    }
+
+    @Override
+    public boolean isValidInput(Object input) {
+        String fileContents = null;
+        String inputCharset = "UTF-8";
+
+        Path path;
+
+        if (input instanceof Path) {
+            path = (Path) input;
+        } else if (input instanceof File) {
+            path = ((File) input).toPath();
+        }
+
+        else {
+            try {
+                // Input corresponds to the parsed file
+                Map<String, Object> mapModel = createModel(input);
+                mapModel = (Map<String, Object>) mapModel.get("model");
+                path = Paths.get(mapModel.get("path").toString());
+                return isValidInput(path);
+            } catch (NullPointerException e) {
+                return false;
+            }
+
+        }
+
+        if (request.isNotConnected()) {
+            startServerConnection();
+        }
+
+        // File content is not needed, as only the file extension is checked
+        fileContents = new String("");
+        // String fileName = path.getFileName().toString();
+        String fileName = path.toString();
+        InputFileTo inputFile = new InputFileTo(fileName, fileContents, inputCharset);
+
+        HttpURLConnection conn = request.getConnection("POST", "Content-Type", "application/json", "isValidInput");
+
+        if (request.sendRequest(inputFile, conn, "UTF-8")) {
+
+            StringBuffer response = new StringBuffer();
+            try (InputStreamReader isr = new InputStreamReader(conn.getInputStream());
+                BufferedReader br = new BufferedReader(isr);) {
+
+                LOG.info("Receiving response from Server....");
+                Stream<String> s = br.lines();
+                s.parallel().forEachOrdered((String line) -> {
+                    response.append(line);
+                });
+                return Boolean.parseBoolean(response.toString());
+            } catch (NullPointerException e) {
+                return false;
+
+            } catch (IOException e) {
+                connectionExc.handle(e);
+                e.printStackTrace();
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public Map<String, Object> createModel(Object input) {
+        Map<String, Object> pojoModel = new HashMap<>();
+
+        try {
+
+            ObjectMapper mapper = new ObjectMapper();
+            String json = input.toString();
+
+            // convert JSON string to Map
+            pojoModel.put("model", mapper.readValue(json, new TypeReference<Map<String, Object>>() {
+            }));
+
+            return pojoModel;
+
+        } catch (JsonGenerationException e) {
+            e.printStackTrace();
+        } catch (JsonMappingException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Can you add some good error messages here? For instance `LOG.error(JsonMappingException exception while trying to parse the model into a map)`